airbyte-commons/src/main/java/io/airbyte/commons/json/Jsons.java

```diff
@@ -129,6 +129,17 @@ public static byte[] toBytes(final JsonNode jsonNode) {
     return serialize(jsonNode).getBytes(Charsets.UTF_8);
   }
 
+  /**
+   * Use string length as an estimation for byte size, because all ASCII characters are one byte long
+   * in UTF-8, and ASCII characters cover most of the use cases. To be more precise, we can convert
+   * the string to byte[] and use the length of the byte[]. However, this conversion is expensive in
+   * memory consumption. Given that the byte size of the serialized JSON is already an estimation of
+   * the actual size of the JSON object, using a cheap operation seems an acceptable compromise.
+   */
+  public static int getEstimatedByteSize(final JsonNode jsonNode) {
+    return serialize(jsonNode).length();
+  }
```

Comment:
Since we are already playing the game of memory optimization, and we are comfortable with estimates, is there a way we can get the size of the object without re-serializing the instance back to a string again?  That's also got overhead...  

A rough approximation could be to store a constant for how much memory an empty JsonNode takes in bytes, and subtract that from this instance's size maybe?

Alternatively, before we parse the message, could we get the string length then and store that value as a property of `AirbyteMessage`... perhaps `AirbyteMessage.originalMessageSize`

Reply:
> A rough approximation could be to store a constant for how much memory an empty JsonNode takes in bytes, and subtract that from this instance's size maybe?

By "this instance's size", do you mean the serialized byte size or the size in memory of a `JsonNode` instance? There is no good way to measure an object's size in memory though. The libraries that can do this usually have large overhead, and are not recommended for production.

It's possible to implement one by tracking the JSON field and value sizes in the `JsonNode` object itself. However, that requires forking the jackson library and more complicated work.

>  Alternatively, before we parse the message, could we get the string length then and store that value as a property of `AirbyteMessage`... perhaps `AirbyteMessage.originalMessageSize`

That's a good idea. Somewhere in the source, it's probably serializing the data inside message already. So we can store the serialized string size as a field in `AirbyteMessage`. However, I cannot find where we construct the `AirbyteMessage` in the upstream.

I will merge this PR as is for now.